content/library/advanced-features/app-testing/get-started.md

```diff
@@ -0,0 +1,267 @@
+   ```
+3. Simulate a user clicking the "**Add**" button (and the resulting script rerun).
+   ```python
+   at.button[0].click().run()
+   ```
+4. Check if the correct message is displayed at the end.
+   ```python
+   assert at.markdown[0].value == "Toe beans counted: 1"
+   ```
+
+Assertions are the heart of tests. When the assertion is true, the test passes. When the assertion is false, the test fails. A test can have multiple assertions, but it is a good practice to keep tests tightly focused. When tests focus on a single behavior, it is easier to understand and respond to failure.
+
+### Try out a simple test with `pytest`
+
+1. Install `pytest` into your Streamlit development environment if you haven't already done so:
+   ```bash
+   pip install pytest
+   ```
+2. Copy the files above into a new "myproject" directory.
+3. Open a terminal and change directory to your project.
+   ```bash
+   cd myproject
+   ```
+4. Execute `pytest`:
+   ```bash
+   pytest
+   ```
+
+The test should execute successfully and your terminal should show something like this:
+
+![A successfully completed test using pytest](/images/app-testing-pytest-intro.png)
+
+By executing `pytest` at the root of your project directory, all Python files with the test prefix (`test_<name>.py`) will be scanned for test functions. Within each test file, each function with the test prefix will be executed as a test. You can also direct `pytest` to only scan your testing directory. For example, from the root of your project directory, execute:
+
+```bash
+pytest tests/
+```
+
+### Handling file paths and imports with `pytest`
+
+Imports and paths within a test script should be relative to the directory where `pytest` is called. That is why the test function uses the path `app.py` instead of `../app.py` even though the app file is one directory up from the test script. In most cases, you'll call `pytest` from the directory containing your main app file. This is typically the root of your project directory.
+
+Additionally, if `.streamlit/` is present in the directory where you call `pytest`, any `config.toml` and/or `secrets.toml` within it will be accessible to your simulated app. For example, your simulated app will have access to the `config.toml` and `secrets.toml` files in this common setup:
+
+Project structure:
+
+```none
+myproject/
+├── .streamlit/
+│   ├── config.toml
+│   └── secrets.toml
+├── app.py
+└── tests/
+    └── test_app.py
+```
+
+Initialization within `test_app.py`:
+
+```python
+# Path to app file is relative the myproject/
+at = AppTest.from_file("app.py").run()
+```
+
+Command to execute tests:
+
+```bash
+cd myproject
+pytest tests/
+```
+
+## Fundamentals of app testing
+
+Now that you understand the basics of `pytest`, let's dive into using Streamlit's app testing framework. Every test begins with initializing and running your simulated app. Additional commands are used to retrieve, manipulate, and inspect elements.
```

Comment:
Should it be:

> Now that you understand the basics of AppTest and `pytest`, ...

Reply:
I mean for the first section to set the context of pytest and merely foreshadow the basics that are covered from this point on.